src/datamodel.js

```diff
@@ -545,75 +545,75 @@ class DataModel extends Relation {
      *  const binDM = dataModel.bin('horsepower', config);
      *
      * @public
      *
-     * @param {String} name Name of measure which will be used to create bin
-     * @param {Object} config Config required for bin creation
-     * @param {Array.<Number>} config.bucketObj.stops Definition of bucket ranges. Two subsequent number from arrays
-     *      are picked and a range is created. The first number from range is inclusive and the second number from range
-     *      is exclusive.
-     * @param {Number} [config.bucketObj.startAt] Force the start of the bin from a particular number.
-     *      If not mentioned, the start of the bin or the lower domain of the data if stops is not mentioned, else its
-     *      the first value of the stop.
-     * @param {Number} config.binSize Bucket size for each bin
-     * @param {Number} config.binCount Number of bins which will be created
-     * @param {String} config.name Name of the new binned field to be created
-     *
-     * @returns {DataModel} Instance of new DataModel with the newly created bin.
+     * @param {string} measureFieldName - The name of the target measure field.
+     * @param {Object} config - The config object.
+     * @param {string} [config.name] - The name of the new field which will be created.
+     * @param {string} [config.buckets] - An array containing the bucket ranges.
+     * @param {string} [config.binSize] - The size of each bin. It is ignored when buckets are given.
+     * @param {string} [config.binsCount] - The total number of bins to generate. It is ignored when buckets are given.
+     * @param {string} [config.start] - The start value of the bucket ranges. It is ignored when buckets are given.
+     * @param {string} [config.end] - The end value of the bucket ranges. It is ignored when buckets are given.
+     * @return {DataModel} Returns a new {@link DataModel} instance with the new field.
      */
-    bin (dimensionName, config = {}) {
-        const clone = this.clone();
-        const binFieldName = config.name || `${dimensionName}_binned`;
-        if (this.getFieldsConfig()[binFieldName] || !this.getFieldsConfig()[dimensionName]) {
-            throw new Error(`Field ${dimensionName} already exists.`);
+    bin (measureFieldName, config) {
+        if (!this.getFieldsConfig()[measureFieldName]) {
+            throw new Error(`Field ${measureFieldName} doesn't exist`);
+        }
+
+        const binFieldName = config.name || `${measureFieldName}_binned`;
+
+        if (this.getFieldsConfig()[binFieldName]) {
+            throw new Error(`Field ${binFieldName} already exists`);
         }
-        const field = this._partialFieldspace.fields.find(currfield => currfield.name() === dimensionName);
-        const dataSet = createBinnedFieldData(field, this._rowDiffset, config);
-        const binField = createFields([dataSet.data], [
+
+        const measureField = this.getFieldspace().fields.find(f => f.name() === measureFieldName);
```

Comment:
Instead of searching in array ```this.getFieldspace().fieldsObj()``` can be used to get the field instance

Reply:
this is old code

Reply:
This is old code.